### PR TITLE
Use Set to allocate CounterID .

### PR DESCRIPTION
**Pull request by Adityav369**

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: pfcpiface/up4.go [hunk 1]
```diff
@@ -59,9 +57,10 @@ type application struct {
 }
 
 type counter struct {
-	maxSize   uint64
-	counterID uint64
-	allocated map[uint64]uint64
+	maxSize        uint64
+	counterID      uint64
+	counterIDsPool set.Set
+	allocated      map[uint64]uint64
```
Comment (osinstom) [inline pfcpiface/up4.go:63]: Do we still need `allocated` map ?

File: pfcpiface/up4.go [hunk 2]
```diff
@@ -162,31 +166,19 @@ func resetCounterVal(p *UP4, counterID uint8, val uint64) {
 }
 
 func (up4 *UP4) getCounterVal(counterID uint8) (uint64, error) {
```
Comment (osinstom) [inline pfcpiface/up4.go:168]:
```suggestion
func (up4 *UP4) allocateCounterID(p4CounterID uint8) (uint64, error) {
```

**Review by osinstom — COMMENTED** [review]
I don't see anything related to releasing counter IDs. See `resetCounterVal`.

File: pfcpiface/up4.go [hunk 3]
```diff
@@ -158,35 +161,23 @@ func (up4 *UP4) initCounter(counterID uint8, name string) error {
 
 func resetCounterVal(p *UP4, counterID uint8, val uint64) {
```
Comment (osinstom) [inline pfcpiface/up4.go:162]:
```suggestion
func (up4 *UP4) releaseCounterID(p4CounterID uint8, val uint64) {
```

**Review by osinstom — APPROVED** [review]
LGTM, just a minor comment

File: pfcpiface/up4.go (hunk 3 above)
Comment (Adityav369) [reply pfcpiface/up4.go]: Done.

File: pfcpiface/up4.go [hunk 4]
```diff
@@ -156,37 +159,25 @@ func (up4 *UP4) initCounter(counterID uint8, name string) error {
 	return nil
 }
 
-func resetCounterVal(p *UP4, counterID uint8, val uint64) {
+func releaseCounterID(up4 *UP4, p4counterID uint8, val uint64) {
```
Comment (osinstom) [inline pfcpiface/up4.go:162]: @Adityav369 can you use `up4 *UP4` as a function receiver, instead of as a function argument?

Comment (Adityav369) [reply pfcpiface/up4.go]: Done.

**Review by osinstom — APPROVED** [review]

**Pull request merged**